src/sage/groups/generic.py

```diff
@@ -1481,12 +1483,18 @@ def has_order(P, n, operation='+'):
         In some cases, order *testing* can be much faster than
         *computing* the order using :func:`order_from_multiple`.
     """
+    from sage.arith.misc import factor
+
     if isinstance(n, sage.rings.integer.Integer):
         if n <= 0:
             return False
         n = n.factor()
+    elif isinstance(n, int):
+        if n <= 0:
+            return False
+        from sage.structure.coerce import py_scalar_to_element
+        n = py_scalar_to_element(n).factor()
 
-    G = P.parent()
```

Comment:
This cannot be deleted as it is called in the `return _rec(P, n)`.

Reply:
What do you mean?
```python
def _rec(Q, fn):
    if not fn:
        return isid(Q)
                                       
    if len(fn) == 1:
        p, k = fn[0]
        for _ in range(k):
            if isid(Q):
                return False
            Q = mult(Q, p)
        return isid(Q)
                                       
    fl = fn[::2]
    fr = fn[1::2]
    l = prod(p**k for p,k in fl)
    r = prod(p**k for p,k in fr)
    L, R = mult(Q, r), mult(Q, l)
    return _rec(L, fl) and _rec(R, fr)
```
doesn't call `G` anywhere, and my linter says it's unused as well.

Reply:
Sorry, I misread the `P` as `G` down below.